File: src/main/java/com/janrain/backplane/server2/BusOwner.java
package com.janrain.backplane.server2;

import com.janrain.backplane.common.User;

/**
 * @author Johnny Bufu
 */
public class BusOwner extends User {

    public BusOwner() { }

    // PACKAGE

    BusOwner(String name, String secret) {
        super(name, secret);
    }

    // PRIVATE

    private static final long serialVersionUID = -6311958895910030506L;

}

File: src/main/scala/com/janrain/backplane/server2/model/BusOwner.scala
package com.janrain.backplane.server2.model

import com.janrain.backplane.common.model.{UserFieldEnum, User}

/**
 * @author Johnny Bufu
 */
class BusOwner(data: Map[String,String]) extends User("bp2BusOwner", data, BusOwnerFields.values) {

  def this(username: String, pwdhash: String) = this(Map(
    BusOwnerFields.USER.name -> username,
    BusOwnerFields.PWDHASH.name -> pwdhash
  ))

  def idField = BusOwnerFields.USER
}

object BusOwnerFields extends UserFieldEnum {
  trait BusOwnerEnumVal extends EnumVal
  // no new, BusOwner.-specific fields
}
